mangoPURE/models/metrics.py

```diff
@@ -0,0 +1,30 @@
+from .base import TimedLoss
+import torch
+from ..data.utils import TimedAudioBatch
+
+
+class SigmoidTimedLoss(TimedLoss):
+    def __init__(self, info_type: str = "noise"):
+        """
+        :param info_type: ["noise", "speaker"] selects which metadata from batch to use
+        """
+        self.info_type = info_type
+
+    def __call__(self, batch: TimedAudioBatch) -> torch.Tensor:
+        mask = torch.zeros(batch.audio.shape)
+        for i, length in enumerate(batch.lengths):
```

Comment:
for is slow af , better write alchemic operation mask[magic(batch.lengths)] (ask got)